### ISSUE TITLE
Building a publishable library that is extending another publishable library (shared library).

### ISSUE THREAD

**Issue by zwarag** [nrwl/nx#5840]
I'm having the troubles when building a publishable lib that is extending another publishable lib.
Setup:
 - dev-app (angular app)
 - lib-core (publishable lib)
 - lib-extend(publishable lib that extends lib-core)

![image](https://user-images.githubusercontent.com/2607973/120173845-bfe44d00-c204-11eb-8ad9-67005313eeab.png)

I read these issues to make sure this is not a dupe: #3518 #2794 #602  

Repo to reproduce: [GitHub](https://github.com/zwarag/nx-poc) or [GitHub1s](https://github1s.com/zwarag/nx-poc/)

I've also first asked this question on [slack](https://nrwlcommunity.slack.com/archives/CMFKWPU6Q/p1622119174180400) to make sure it's not a trivial thing that I'm missunderstanding.

I assume I'm doing someting wrong and expecially after reading #3518 I suspect my config to be wrong. However, I cannot figure out what exactly is wrong and maybe it is a edge case that is not "correctly" handeld yet.  

**One more thing**, the libs are written in a way to support secondary entry points. So maybe thats the root of the issue.

## Current Behavior
The command `nx run lib-extend:build --with-deps` returns an error.

## Expected Behavior
It should build the lib-extend library without erroring out.  
For context, it never worked for me.

However, I have two idea of what could be the cause of the error:
1. Wrong usage of the "paths" in [tsconfig.base.json](https://github.com/zwarag/nx-poc/blob/master/tsconfig.base.json#L20)
2. Missing dependency in [{ng-}](https://github.com/zwarag/nx-poc/blob/master/libs/lib-extend/ng-package.json), [package.json](https://github.com/zwarag/nx-poc/blob/master/libs/lib-extend/package.json)? I think I need to add lib-core as a dependency, but my tryings so far did not help.

## Steps to Reproduce
I can run `nx run dev-app:build` and `nx run lib-core:build` without any problems. However, when I run `nx run lib-extend:build --with-deps` I get the error in section "Failure Logs":

### Failure Logs
```
> Executing task: ng build lib-extend --with-deps --skip-nx-cache <


>  NX  Running target build for project lib-extend and 1 task(s) that it depends on.
        

———————————————————————————————————————————————

> nx run lib-core:build 
Building Angular Package

------------------------------------------------------------------------------
Building entry point '@my-org/lib-core'
------------------------------------------------------------------------------
✔ Compiling TypeScript sources through NGC
✔ Bundling to FESM2015
✔ Bundling to UMD
✔ Writing package metadata
✔ Built @my-org/lib-core

------------------------------------------------------------------------------
Building entry point '@my-org/lib-core/button'
------------------------------------------------------------------------------
✔ Compiling TypeScript sources through NGC
✔ Bundling to FESM2015
✔ Bundling to UMD
✔ Writing package metadata
✔ Built @my-org/lib-core/button

------------------------------------------------------------------------------
Building entry point '@my-org/lib-core/unordered-list'
------------------------------------------------------------------------------
✔ Compiling TypeScript sources through NGC
✔ Bundling to FESM2015
✔ Bundling to UMD
✔ Writing package metadata
✔ Built @my-org/lib-core/unordered-list

------------------------------------------------------------------------------
Built Angular Package
 - from: /home/wrskah/repos/org/libs/lib-core
 - to:   /home/wrskah/repos/org/dist/@my-org/lib-core
------------------------------------------------------------------------------

Build at: 2021-05-31T09:34:43.136Z - Time: 2766ms


> nx run lib-extend:build 
Some of the project lib-extend's dependencies have not been built yet. Please build these libraries before:
- lib-core

Try: nx run lib-extend:build --with-deps

———————————————————————————————————————————————

>  NX   ERROR  Running target "lib-extend:build" failed

  Failed tasks:
  
  - lib-extend:build

The terminal process "/usr/bin/bash '-l', '-c', '/home/wrskah/repos/org/node_modules/.bin/ng build lib-extend --with-deps --skip-nx-cache', 'ng build lib-extend --with-deps --skip-nx-cache'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

### Environment
```
➜  org git:(master) nx report

>  NX  Report complete - copy this into the issue template

  Node : 12.22.1
  OS   : linux x64
  yarn : 1.22.10
  
  nx : Not Found
  @nrwl/angular : 12.3.5
  @nrwl/cli : 12.3.5
  @nrwl/cypress : 12.3.5
  @nrwl/devkit : 12.3.5
  @nrwl/eslint-plugin-nx : 12.3.5
  @nrwl/express : Not Found
  @nrwl/jest : 12.3.5
  @nrwl/linter : 12.3.5
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.3.5
  @nrwl/web : Not Found
  @nrwl/workspace : 12.3.5
  @nrwl/storybook : Not Found
  @nrwl/gatsby : Not Found
  typescript : 4.2.4
```

**Comment by zwarag** [comment]
After further investigation I've found a solution. I suggest this should be communicated in the docs and/or it potentially could be a bug.

juristr pointed out [here](https://github.com/nrwl/nx/issues/2479#issuecomment-756796806) that if the `dest` path is changed inside the ng-package, then it is necessary to provide the `outputs` property in the angular.json file.

I cannot remember changing the `dest` path and had to provide `outputs` to make it work.

----

For others stumbling upon this problem:
The library `lib-extend` extends the `lib-core` library. I had to add [this line]( https://github.com/zwarag/nx-poc/blob/78f869054ffe2fb8ff2909dc70b8f89ce70819d2/angular.json#L169) to the `lib-core`, so that `lib-extend` could be build without an error.

----

For my part this issue is resolved. I'll let it open in case it is relevant for nrwl team. If you're a dev/community manager and feel like this needs no further investigation, you can close it.

And a thank you for the awesome work 👍. I really enjoy nx!

**Comment by zwarag** [comment]
For my part this issue is resolved. I'll let it open in case it is relevant for nrwl team. If you're a dev/community manager and feel like this needs no further investigation, you can close it.

And a thank you for the awesome work 👍. I really enjoy nx!

**Comment by leosvelperez** [comment]
> I cannot remember changing the `dest` path and had to provide `outputs` to make it work.

When generating a library, there was an issue where Nx was not generating the `outputs` property. This was resolved [here](https://github.com/nrwl/nx/pull/5867) and released as part of [12.3.6](https://github.com/nrwl/nx/releases/tag/12.3.6). Any library created before that version has this problem. That's why, even though you didn't change the `dest` path, you had to provide the `outputs` property because it wasn't even set before.

**Comment by leosvelperez** [comment]
@zwarag I amended [the docs](https://nx.dev/latest/angular/ci/setup-incremental-builds-angular#use-buildable-libraries) and added a notice about keeping the `outputs` property in sync with the `dest` property in `ng-package.json`.

I'm closing this one since it has been solved and the docs have been updated with more info.
